packages/service-library/src/servicelib/redis.py

```diff
@@ -220,20 +218,25 @@ class RedisClientsManager:
     Manages the lifetime of redis client sdk connections
     """
 
-    databases: set[RedisDatabase]
+    databases_configs: set[RedisManagerDBConfig]
     settings: RedisSettings
 
     _client_sdks: dict[RedisDatabase, RedisClientSDK] = field(default_factory=dict)
 
     async def setup(self) -> None:
-        for db in self.databases:
-            self._client_sdks[db] = client_sdk = RedisClientSDK(
-                redis_dsn=self.settings.build_redis_dsn(db)
+        for config in self.databases_configs:
+            self._client_sdks[config.database] = RedisClientSDK(
+                redis_dsn=self.settings.build_redis_dsn(config.database),
+                decode_responses=config.decode_responses,
+                health_check_interval=config.health_check_interval,
             )
-            await client_sdk.setup()
+
+        for client in self._client_sdks.values():
+            await client.setup()
 
     async def shutdown(self) -> None:
-        await logged_gather(*(c.shutdown() for c in self._client_sdks.values()))
+        for client in self._client_sdks.values():
```

Comment:
why not keeping going with the logged_gather here?

Reply:
I'm gonna put a note here. If I do such a thing it will hang. Not sure what is happening.